testing: fakePoS module: add control for stop/start of subscription event loop

nonsense opened the pull request:
**Description**

This PR is modifying the `fakePoS` module within Geth, and is adding a control API so that we can stop/start it on-demand from tests.

**Additional context**

Necessary for https://github.com/ethereum-optimism/optimism/issues/14128

nonsense committed:
control-plane: add fakePoS control within L1CL

pcw109550 reviewed (commented):
Can we add a test for validating the FakePoS controller at https://github.com/ethereum-optimism/optimism/blob/develop/op-devstack/sysgo/control_plane_test.go ?

Also some suggestion:

Current `fakePoS` struct implements a `Lifecycle` interface:
```go
type Lifecycle interface {
	// Start is called after all services have been constructed and the networking
	// layer was also initialized to spawn any goroutines required by the service.
	Start() error

	// Stop terminates all goroutines belonging to the service, blocking until they
	// are all terminated.
	Stop() error
}
```
Is there any difference between pausing using the `ctrl   chan FakePoSRequest` channel, and the `fakePos.Stop() method`? If not, we may simply call these start/stop methods. 

https://github.com/ethereum-optimism/optimism/blob/4a6b9ee62cafc490b9a296bbfc5e62f865fd2c45/op-e2e/e2eutils/geth/geth.go#L63-L72

So like
```
	fakePoS := &fakePoS{
		clock:             c,
		eth:               gethInstance.Backend,
		log:               log.Root(), // geth logger is global anyway. Would be nice to replace with a local logger though.
		blockTime:         blockTime,
		finalizedDistance: finalizedDistance,
		safeDistance:      4,
		engineAPI:         catalyst.NewConsensusAPI(gethInstance.Backend),
		beacon:            beaconSrv,
	}
	gethInstance.Node.RegisterLifecycle(fakePoS)

         // 	fakePoS.Stop(),  	fakePoS.Start()
```

Expose these methods by adding methods at `GethInstance` struct.

nonsense commented on op-e2e/e2eutils/geth/fakepos.go:
```diff
@@ -212,9 +226,13 @@ func (f *fakePoS) Start() error {
-	f.sub.Unsubscribe()
-	if advancing, ok := f.clock.(*clock.AdvancingClock); ok {
-		advancing.Stop()
+	if f.sub != nil {
```
Looks like our tests are calling `Stop` before we have ever called `Start`, so we need `nil` checks.

nonsense commented:
@pcw109550 thanks for review. Indeed we can call Start/Stop, so I modified our ControlPlane to use them directly.

I've added a test, separate from the other control plane tests, so that they don't interfere. I don't think we want to stop the L1 chain, while other tests are in-flight.

nonsense committed:
fix comment

pcw109550 commented on op-e2e/e2eutils/geth/fakepos.go:
```diff
@@ -26,6 +26,20 @@ type Beacon interface {
 
+// FakePoS is a testing-only utility to attach to Geth
+// the exported struct is used by the devstack to control the fakePoS module, and adheres to the devstack.Lifecycle interface
+type FakePoS struct {
```
I think we are safe to expose the original `fakePoS` struct, not wrapping around additional boilerplate struct.

pcw109550 commented on op-e2e/e2eutils/geth/fakepos.go:
```diff
@@ -26,6 +26,20 @@ type Beacon interface {
+}
+
+func (f *FakePoS) Start() {
+	_ = f.fakePoS.Start()
```
Suggestion: These errors get lost, and not logged when error. Currently L1CLNode does not have a logger field to do so. Shall we add a devtest.P field to L1CLNode and use this for log when error? Also, I think we need a separate sysgo struct for this: for example at sysgo/l1_nodes.go

```go
struct FakePoS {
    p devtest.P
    fakepos        *geth.FakePoS
}

func (f *FakePoS) Start() {
    f.p.Require(f.fakePoS.Start(), "fakePoS failed to start")
}


func (f *FakePoS) Stop() {
    f.p.Require(f.fakePoS.Stop(), "fakePoS failed to stop")
}
```
and plug in the sysgo fakePoS to sysgo L1CLNode.

pcw109550 commented on op-e2e/e2eutils/geth/fakepos.go:
```diff
@@ -212,9 +226,13 @@ func (f *fakePoS) Start() error {
+			advancing.Stop()
+		}
 	}
 	return nil
```
We may return an error when necessary fields are nil when `Stop` is called.

nonsense committed:
return err when calling Stop while fakePoS is not initialised

nonsense committed:
move FakePoS under sysgo package
